pavelib/prereqs.py

```diff
@@ -121,23 +124,49 @@ def prereq_cache(cache_name, paths, install_func):
         print '{cache} unchanged, skipping...'.format(cache=cache_name)
 
 
+@timeout(limit=60)
 def node_prereqs_installation():
     """
     Configures npm and installs Node prerequisites
     """
+    # NPM installs hang sporadically. Log the installation process so that we
+    # determine if any packages are chronic offenders.
+    shard_str = os.getenv('SHARD', None)
+    if shard_str:
+        npm_log_file_path = '{}/npm-install.{}.log'.format(Env.GEN_LOG_DIR, shard_str)
+    else:
+        npm_log_file_path = '{}/npm-install.log'.format(Env.GEN_LOG_DIR)
+    npm_log_file = open(npm_log_file_path, 'w')
```

Comment:
Reminder: the `open()` built-in function goes away in Python 3, replaced by `io.open()`